AzureCP/AzureCP.cs

```diff
@@ -491,24 +491,83 @@ public void BuildFilterAndProcessResultsAsync(string input, List<AzureADObject>
                 }
             }
 
-            Task<List<AzurecpResult>> searchResultsTask = this.QueryAzureADCollectionAsync(input, userQuery, groupQuery);
+            // search users, groups and domains
+            Task<List<object>> searchResultsTask = this.QueryAzureADCollectionAsync(input, userQuery, groupQuery);
             searchResultsTask.ConfigureAwait(false);
+            // wait for task
             searchResultsTask.Wait();
-            List<AzurecpResult> searchResults = searchResultsTask.Result;
-            if (searchResults == null || searchResults.Count == 0) return;
+            // Separate AzurecpResult from string (domains)
+            IEnumerable<AzurecpResult> searchResults = searchResultsTask.Result.OfType<AzurecpResult>();
+            IEnumerable<string> domains = searchResultsTask.Result.OfType<string>();
```

Comment:
Current solution works great, but as discussed earlier, it's really not elegant. 
I like your suggestion to split domains into a separate list. Do you plan to do it?